### ISSUE TITLE
guide on how to run it on windows

### ISSUE THREAD

**Issue by mhnoni** [ShivamShrirao/diffusers#56]
can't make it run on windows natively, getting this error after running **python train_dreambooth.py**

train_dreambooth.py: error: the following arguments are required: --pretrained_m
odel_name_or_path, --pretrained_vae_name_or_path, --instance_data_dir

maybe I'm doing something wrong? the only difference I made is to run python train_dreambooth.py instead of launch.sh for Linux.

**Comment by NeoAnthropocene** [comment]
> can't make it run on windows natively, getting this error after running **python train_dreambooth.py**
> 
> train_dreambooth.py: error: the following arguments are required: --pretrained_m odel_name_or_path, --pretrained_vae_name_or_path, --instance_data_dir
> 
> maybe I'm doing something wrong? the only difference I made is to run python train_dreambooth.py instead of launch.sh for Linux.

@mhnoni 
I would suggest watching the video on the below link and read carefully the updates on the video description. You can run it locally by following the steps carefully in that video.

(https://youtu.be/w6PTviOCYQY)

**Comment by mhnoni** [comment]
> > can't make it run on windows natively, getting this error after running **python train_dreambooth.py**
> > train_dreambooth.py: error: the following arguments are required: --pretrained_m odel_name_or_path, --pretrained_vae_name_or_path, --instance_data_dir
> > maybe I'm doing something wrong? the only difference I made is to run python train_dreambooth.py instead of launch.sh for Linux.
> 
> @mhnoni I would suggest watching the video on the below link and read carefully the updates on the video description. You can run it locally by following the steps carefully in that video.
> 
> (https://youtu.be/w6PTviOCYQY)

That one uses wsl to install Linux. I would like to use it natively on windows without Linux which seems possible but no guide on how to run the last command on windows, I mean instead of launch.sh we should have a python file to run train_dreambooth.py but not idea how to do that.

**Comment by mhnoni** [comment]
> Here you go: https://www.reddit.com/r/StableDiffusion/comments/ydip3s/guide_dreambooth_training_with_shivamshriraos/

oh wow, thank you so much!

**Comment by InB4DevOps** [comment]
@ShivamShrirao can we please have a Wiki in this repo where we could document things like this?

**Comment by ShivamShrirao** [comment]
@InB4DevOps yeah I wanted to but I haven't been able to verify them myself. Like the one linked here doesn't install xformers.  Also the devs for xformers and bitsandbytes have been working on getting them officially working on windows.
Will add a section where I can add these unverified resources until then.

**Comment by InB4DevOps** [comment]
@ShivamShrirao a wiki would be nice to have nonetheless. For all the other knowledge surrounding DreamBooth with SD..

**Comment by mhnoni** [comment]
regarding xformers
from auto111 code: 
pip install https://github.com/C43H66N12O12S2/stable-diffusion-webui/releases/download/c/xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl

needs python 10.
I was able to install but but can't confirm if it will work or not since I only have 8GBVram. waiting for deepspeed.

**Comment by patrickgalbraith** [comment]
This is what I did to get it working.

**environment.yml**

```yaml
name: dbshiv
channels:
  - conda-forge
  - pytorch
  - nvidia
  - defaults
dependencies:
  - python>=3.10
  - pip
  - pytorch==1.12.1
  - cudatoolkit=11.6.0
  - numpy
  - pip:
    - accelerate==0.12.0
    - torchvision==0.13.1
    - torchaudio==0.12.1
    - ftfy
    - tensorboard
    - modelcards
    - bitsandbytes
    - transformers>=4.21.0
    - pyre-extensions==0.0.23
    - -e git+https://github.com/ShivamShrirao/diffusers.git@main#egg=diffusers
```

```bash
# Note the pytorch version used is important since newer versions don't work right now
conda env create -f environment.yml
conda activate dbshiv

# Download https://github.com/C43H66N12O12S2/stable-diffusion-webui/releases/download/f/xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl
pip install -U -I --no-deps xformers-0.0.14.dev0-cp310-cp310-win_amd64.whl

# Run accelerate config
#  In which compute environment are you running? ([0] This machine, [1] AWS (Amazon SageMaker)): 0
#  Which type of machine are you using? ([0] No distributed training, [1] multi-CPU, [2] multi-GPU, [3] TPU [4] MPS): 0
#  Do you want to run your training on CPU only (even if a GPU is available)? [yes/NO]:no
#  Do you want to use DeepSpeed? [yes/NO]: no
#  Do you wish to use FP16 or BF16 (mixed precision)? [NO/fp16/bf16]: FP16
accelerate config

# Make sure GPU is enabled in the follow output, if not then pytorch install is not correct
accelerate env

# To train update and run
train.bat

# To convert to checkpoint
python convert_diffusers_to_original_stable_diffusion.py --model_path="./output/out" --checkpoint_path="./output/checkpoint.ckpt"
```

**Comment by NeoAnthropocene** [comment]
@patrickgalbraith , thanks for sharing this clean code.
I wonder why you are using the LR Scheduler with the  "polynomial". Did you get better results compared with "constant" parameter?

I don't have deep knowledge about it but I'm trying to find out the best results by trying different parameters that I saw from other examples around the internet. I just wanted to know if it worth to spend time on it or not.

Thanks.

**Comment by NeoAnthropocene** [comment]
> 

Thank you so much. This will save so much time :)
